pkg/resources/api_authentication_integration_with_client_credentials.go

```diff
@@ -0,0 +1,486 @@
+		return property.Name == "OAUTH_CLIENT_AUTH_METHOD"
+	})
+	if err == nil {
+		if err = d.Set("oauth_client_auth_method", oauthClientAuthMethod.Value); err != nil {
+			return nil, err
+		}
+	}
+	oauthTokenEndpoint, err := collections.FindOne(properties, func(property sdk.SecurityIntegrationProperty) bool { return property.Name == "OAUTH_TOKEN_ENDPOINT" })
+	if err == nil {
+		if err = d.Set("oauth_token_endpoint", oauthTokenEndpoint.Value); err != nil {
+			return nil, err
+		}
+	}
+	oauthAllowedScopes, err := collections.FindOne(properties, func(property sdk.SecurityIntegrationProperty) bool { return property.Name == "OAUTH_ALLOWED_SCOPES" })
+	if err == nil {
+		if err = d.Set("oauth_allowed_scopes", listValueToSlice(oauthAllowedScopes.Value, true, false)); err != nil {
+			return nil, err
+		}
+	}
+	oauthGrant, err := collections.FindOne(properties, func(property sdk.SecurityIntegrationProperty) bool { return property.Name == "OAUTH_GRANT" })
+	if err == nil {
+		if err = d.Set("oauth_grant", oauthGrant.Value); err != nil {
+			return nil, err
+		}
+	}
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func CreateContextApiAuthenticationIntegrationWithClientCredentials(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+	enabled := d.Get("enabled").(bool)
+	name := d.Get("name").(string)
+	oauthClientId := d.Get("oauth_client_id").(string)
+	oauthClientSecret := d.Get("oauth_client_secret").(string)
+	id := sdk.NewAccountObjectIdentifier(name)
+	req := sdk.NewCreateApiAuthenticationWithClientCredentialsFlowSecurityIntegrationRequest(id, enabled, oauthClientId, oauthClientSecret)
+
+	if v, ok := d.GetOk("comment"); ok {
+		req.WithComment(v.(string))
+	}
+
+	if v := d.Get("oauth_access_token_validity").(int); v != -1 {
+		req.WithOauthAccessTokenValidity(v)
+	}
+
+	if v := d.Get("oauth_client_auth_method").(string); v != "unknown" {
+		value, err := sdk.ToApiAuthenticationSecurityIntegrationOauthClientAuthMethodOption(v)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		req.WithOauthClientAuthMethod(value)
+	}
+
+	if v := d.Get("oauth_refresh_token_validity").(int); v != -1 {
+		req.WithOauthRefreshTokenValidity(v)
+	}
+
+	if v := d.Get("oauth_grant").(string); v != "unknown" {
+		if v == "CLIENT_CREDENTIALS" {
+			req.WithOauthGrantClientCredentials(true)
+		}
+	}
+
+	if v := d.Get("oauth_token_endpoint").(string); v != "unknown" {
+		req.WithOauthTokenEndpoint(v)
+	}
+	if v, ok := d.GetOk("oauth_allowed_scopes"); ok {
+		elems := expandStringList(v.(*schema.Set).List())
+		allowedScopes := make([]sdk.AllowedScope, len(elems))
+		for i := range elems {
+			allowedScopes[i] = sdk.AllowedScope{Scope: elems[i]}
+		}
+		req.WithOauthAllowedScopes(allowedScopes)
+	}
+
+	if err := client.SecurityIntegrations.CreateApiAuthenticationWithClientCredentialsFlow(ctx, req); err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(name)
```

Comment:
You should use EncodeSnowflakeID (it can lead to diffs based on quotes)

Reply:
Done.